src/ui/tray-icon.cpp

```diff
@@ -260,6 +260,12 @@ void SeafileTrayIcon::showMessageWithRepo(const QString& repo_id, const QString
 
 void SeafileTrayIcon::rotateTrayIcon()
 {
+    /* tray icon should not be refreshed on Gnome according to their guidelines */
+    const char *env = g_getenv("DESKTOP_SESSION");
+    if (env && strcmp(env, "gnome") == 0) {
+        return;
+    }
+
```

Comment:
Thanks for the PR! However I think the function `SeafileTrayIcon::rotate` is a better place to put the code. 

https://github.com/haiwen/seafile-client/blob/3829b0a3e731c9d0baa43c60d249422aabd81d80/src/ui/tray-icon.cpp#L213